internal/hcl/funcs/yaml.go

```diff
@@ -0,0 +1,44 @@
+package funcs
+
+import (
+	"strings"
+
+	yaml "github.com/zclconf/go-cty-yaml"
+	"github.com/zclconf/go-cty/cty"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+var YAMLDecodeFunc = function.New(&function.Spec{
+	Params: []function.Parameter{
+		{
+			Name: "src",
+			Type: cty.String,
+		},
+	},
+	Type: func(args []cty.Value) (cty.Type, error) {
+		if !args[0].IsKnown() {
+			return cty.DynamicPseudoType, nil
+		}
+		if args[0].IsNull() {
+			return cty.NilType, function.NewArgErrorf(0, "YAML source code cannot be null")
+		}
+		val := args[0].AsString()
+		if strings.HasPrefix(val, "mock") {
```

Comment:
Is it possible for this value to actually start with `mock` and not be a mock?

Reply:
I guess it is theoretically possible but it's highly unlikely as it would require the first key of a yaml/json file to have `mock:` 

e.g:

```
mock:
   - value
```

So I think it's safe to assume it's our internal mocks - I guess longer term we could change our mock string to `infracostmock-...` which would probably mitigate this worry?